src/NuGet.Core/NuGet.Configuration/Settings/Items/AuthorItem.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Shared;
+
+namespace NuGet.Configuration
+{
+    public sealed class AuthorItem : TrustedSignerItem, IEquatable<AuthorItem>
```

Comment:
why do you need separate class impl for author vs repo signers? Can it be part of `TrustedSignerItem` itself with a type and ServiceIndex being optional for author...

Reply:
I think it is cleaner to make each class have a single responsibility. By making the implementation only one class and validating you are doing some things for authors and other for repositories the code becomes more error-prone. It is cleaner and easier to maintain if we separate that responsibility in two different classes that subclass from the same one.